src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
I think it's fine for (ii) to be in a separate PR.  Should we also remove starting the Delay Timer in `SetStateChild()` and `SetStateRouter()`?  The timer should be started after receiving the Pending Dataset from its Parent.